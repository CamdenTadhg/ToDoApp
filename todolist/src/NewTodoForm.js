import react, {useState} from 'react';

const NewTodoForm = ({addTodo}) => {
    const [task, setTask] = useState('');
    const handleChange = (event) => {
        setTask(event.target.value)
    }
    const handleSubmit = (event) => {
        event.preventDefault();
        addTodo(task);
        setTask('');
    }

    return (
        <form onSubmit={handleSubmit} data-testid="todoform">
            <label htmlFor='todo'>Task: 
                <input type='text' id='todo' name='todo' placeholder='task' value={task} onChange={handleChange}/>
            </label>
            <button>Create ToDo</button>
        </form>
    )
}

export default NewTodoForm;