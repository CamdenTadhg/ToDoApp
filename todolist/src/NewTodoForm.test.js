import react from 'react';
import {render} from '@testing-library/react';
import NewTodoForm from './NewTodoForm.js';

it('should render correctly', function() {
    render(<NewTodoForm/>);
});

it('should match the snapshot', function() {
    const {asFragment} = render(<NewTodoForm/>);
    expect(asFragment()).toMatchSnapshot();
})