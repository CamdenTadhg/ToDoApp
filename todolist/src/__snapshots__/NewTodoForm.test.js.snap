// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <form
    data-testid="todoform"
  >
    <label
      for="todo"
    >
      Task:
      <input
        id="todo"
        name="todo"
        placeholder="task"
        type="text"
        value=""
      />
    </label>
    <button>
      Create ToDo
    </button>
  </form>
</DocumentFragment>
`;
